base/commands/migration/migration_stages.go

```diff
@@ -0,0 +1,328 @@
+		}
+	}
+}
+
+type OverallMigrationStatus struct {
+	Status               Status              `json:"status"`
+	Logs                 []string            `json:"logs"`
+	Errors               []string            `json:"errors"`
+	Report               string              `json:"report"`
+	CompletionPercentage float32             `json:"completionPercentage"`
+	Migrations           []DSMigrationStatus `json:"migrations"`
+}
+
+type DataStructureInfo struct {
+	Name string
+	Type string
+}
+
+type DSMigrationStatus struct {
+	Name                 string  `json:"name"`
+	Type                 string  `json:"type"`
+	Status               Status  `json:"status"`
+	CompletionPercentage float32 `json:"completionPercentage"`
+	Error                string  `json:"error"`
+}
+
+func fetchMigrationStatus(ctx context.Context, ci *hazelcast.ClientInternal, migrationID string) (string, error) {
+	q := fmt.Sprintf(`SELECT JSON_QUERY(this, '$.status') FROM %s WHERE __key='%s'`, StatusMapName, migrationID)
+	res, err := ci.Client().SQL().Execute(ctx, q)
+	if err != nil {
+		return "", err
+	}
+	it, err := res.Iterator()
+	if err != nil {
+		return "", err
+	}
+	if it.HasNext() { // single iteration is enough that we are reading single result for a single migration
+		row, err := it.Next()
+		if err != nil {
+			return "", err
+		}
+		r, err := row.Get(0)
+		var m string
+		if err = json.Unmarshal(r.(serialization.JSON), &m); err != nil {
+			return "", err
+		}
+		return m, nil
+	} else {
+		return "", migrationStatusNotFoundErr
+	}
+	return "", nil
+}
+
+func fetchMigrationReport(ctx context.Context, ci *hazelcast.ClientInternal, migrationID string) (string, error) {
+	q := fmt.Sprintf(`SELECT JSON_QUERY(this, '$.report') FROM %s WHERE __key='%s'`, StatusMapName, migrationID)
+	res, err := ci.Client().SQL().Execute(ctx, q)
+	if err != nil {
+		return "", err
+	}
+	it, err := res.Iterator()
+	if err != nil {
+		return "", err
+	}
+	if it.HasNext() { // single iteration is enough that we are reading single result for a single migration
+		row, err := it.Next()
+		if err != nil {
+			return "", err
+		}
+		r, err := row.Get(0)
+		var m string
+		if err = json.Unmarshal(r.(serialization.JSON), &m); err != nil {
+			return "", err
+		}
+		return m, nil
+	} else {
+		return "", fmt.Errorf("migration report cannot be found")
+	}
+	return "", nil
+}
+
+func fetchMigrationErrors(ctx context.Context, ci *hazelcast.ClientInternal, migrationID string) (string, error) {
+	q := fmt.Sprintf(`SELECT JSON_QUERY(this, '$.errors') FROM %s WHERE __key='%s'`, StatusMapName, migrationID)
+	res, err := ci.Client().SQL().Execute(ctx, q)
+	if err != nil {
+		return "", err
+	}
+	it, err := res.Iterator()
+	if err != nil {
+		return "", err
+	}
+	var errs []string
+	for it.HasNext() {
+		row, err := it.Next()
+		if err != nil {
+			return "", err
+		}
+		r, err := row.Get(0)
+		var m string
+		if err = json.Unmarshal(r.(serialization.JSON), &m); err != nil {
+			return "", err
+		}
```

Comment:
The error returned from `row.Get` is not checked.
Also `serialization.JSON == []byte` so no need to unmarshal it to a string.